webcomponents/tool/src/main/frontend/packages/sakai-push-utils/src/sakai-push-utils.js

```diff
@@ -5,6 +5,102 @@ export const NOT_PUSH_CAPABLE = "NOT_PUSH_CAPABLE";
 
 const pushCallbacks = new Map();
 
+export const isIOSSafari = () => {
+  const ua = navigator.userAgent;
+  return /iPad|iPhone|iPod/.test(ua) && /Safari/.test(ua) && !/Chrome/.test(ua);
+};
+
+export const isAndroid = () => {
+  return /Android/.test(navigator.userAgent);
+};
+
+export const isPWA = () => {
+  return window.matchMedia("(display-mode: standalone)").matches;
+};
+
+export const getBrowserInfo = () => {
```

Comment:
Why not try and use an existing npm module for this?

https://www.npmjs.com/package/ua-parser-js

Reply:
a) license fails
b) i think npm just add a module is a bit of an anti-pattern

but i'm all ears if you have another npm module with a proper license and no crazy transient dependencies